File: 백준/Silver/11723. 집합/집합.py
from sys import stdin
input = stdin.readline

m = int(input())
s = set()
for _ in range(m):
    inp = input().rstrip().split()
    if len(inp) == 1:
        if inp[0] == 'all':
            s = set([i for i in range(1, 21)])
        else:
            s = set()
    else:
        command = inp[0] 
        n = int(inp[1])
        if command == 'add':
            s.add(n)
        elif command == 'remove':
            s.discard(n)
        elif command == 'check':
            if n in s:
                print(1)
            else:
                print(0)
        elif command == 'toggle':
            if n in s:
                s.discard(n)
            else:
                s.add(n)

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine());
        int flag = 0;
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            int x;
            switch (command) {
                case "add":
                    x = Integer.parseInt(st.nextToken());
                    flag |= (1 << x);
                    break;
                case "remove":
                    x = Integer.parseInt(st.nextToken());
                    flag &= ~(1 << x);
                    break;
                case "check":
                    x = Integer.parseInt(st.nextToken());
                    if ((flag & (1 << x)) != 0) sb.append(1).append("\n");
                    else sb.append(0).append("\n");
                    break;
                case "toggle":
                    x = Integer.parseInt(st.nextToken());
                    flag ^= (1 << x);
                    break;
                case "all":
                    for (int j = 1; j <= 20; j++) {
                        flag |= (1 << j);
                    }
                    break;
                case "empty":
                    for (int j = 1; j <= 20; j++) {
                        flag &= ~(1 << j);
                    }
                    break;
            }
        }
        System.out.print(sb);
    }
}
